rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/explorer/MetaTracker.kt

```diff
@@ -30,109 +32,105 @@ import kotlin.io.path.ExperimentalPathApi
-                            val origin = VfsUtil.findFile(metaFile, true)
-                            val conflictingMeta = origin?.parent?.findChild(target)
-                            if (conflictingMeta != null) {
-                                logger.warn("Removing conflicting meta $conflictingMeta")
-                                conflictingMeta.delete(this)
+                        is VFilePropertyChangeEvent -> {
+                            if (!event.isRename) continue
+                            val metaFile = getMetaFile(event.oldPath) ?: continue
+                            actions.add(metaFile) {
+                                val target = getMetaFileName(event.newValue as String)
+                                val origin = VfsUtil.findFile(metaFile, true)
+                                val conflictingMeta = origin?.parent?.findChild(target)
+                                if (conflictingMeta != null) {
+                                    logger.warn("Removing conflicting meta $conflictingMeta")
+                                    conflictingMeta.delete(this)
+                                }
+                                origin?.rename(this, target)
                             }
-                            origin?.rename(this, target)
                         }
                     }
+                } catch (t: Throwable) {
+                    logger.error(t)
+                    continue
                 }
-            } catch (t: Throwable) {
-                logger.error(t)
-                continue
             }
-        }
 
-        if (actions.isEmpty()) return
+            if (actions.isEmpty()) continue
 
-        val commandProcessor = CommandProcessor.getInstance()
-        var groupId = commandProcessor.currentCommandGroupId
-        if (groupId == null) {
-            groupId = MetaGroupId(nextGroupIdIndex++)
-            commandProcessor.currentCommandGroupId = groupId
-        }
+            val commandProcessor = CommandProcessor.getInstance()
+            var groupId = commandProcessor.currentCommandGroupId
+            if (groupId == null) {
+                groupId = MetaGroupId(nextGroupIdIndex++)
+                commandProcessor.currentCommandGroupId = groupId
+            }
 
-        application.invokeLater {
-            commandProcessor.allowMergeGlobalCommands {
-                commandProcessor.executeCommand(project, {
-                    actions.executeUnderWriteLock()
-                }, actions.getCommandName(), groupId)
+            application.invokeLater {
+                commandProcessor.allowMergeGlobalCommands {
+                    commandProcessor.executeCommand(project, {
+                        actions.executeUnderWriteLock()
+                    }, actions.getCommandName(), groupId)
+                }
             }
+            return // action was handled, so we should not try to process it again with other projects
```

Comment:
There's an edge case here, if multiple Unity projects are open and you somehow get events from both projects. This is a pretty small edge case though. I'm finding it hard to think of a good scenario.

Perhaps a `file:` package added to both Project A and Project B, and a bulk update is done for files in project B. Project A would process the shared file, then return. The changes for other files in project B would be missed. I think we'd be fine with this, unless you can think of a more likely scenario?

Reply:
I suppose this scenario would be fine, since we process each event separately for every project opened.

Reply:
The service is application level, it gets all the changes and trues to find appropriate unity project (solution) for each change. Once the event is processed with any of the projects, it stops trying the others.

Reply:
> I suppose this scenario would be fine, since we process each event separately for every project opened.

It looks like it's processing a list of events by looping over each project, and if that project processes at least one event, then it stops looping. This relies on the list of events all belonging to one project, doesn't it?

Reply:
You are right, it looks like a bug. investigating...

Reply:
@citizenmatt Please take a look at this new pull request https://github.com/JetBrains/resharper-unity/pull/2208/files